File: python/121 - Best Time to Buy and Sell Stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        # index of variables
        low = 0
        high = 0
        profit = 0
        
        i = 1
        while i < len(prices):
            if prices[low] > prices[i]:
                low = i
            elif prices[i] - prices[low] > profit:
                high = i
                profit = prices[i] - prices[low]
            i += 1
        return profit

File: cpp/121 - Best Time to Buy and Sell Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        // var
        int profit = 0;
        int min_val = INT_MAX;
        
        for (int i = 0; i < prices.size(); i++) {
            min_val = min(min_val, prices[i]);
            profit = max(profit, prices[i] - min_val);
        }
        return profit;
    }
};